R/ui.R

```diff
@@ -37,24 +40,47 @@ style_pkg <- function(pkg = ".",
                       ...,
                       style = tidyverse_style,
                       transformers = style(...),
+                      filetype = "R",
                       exclude_files = "R/RcppExports.R") {
   pkg_root <- rprojroot::find_package_root_file(path = pkg)
-  changed <- withr::with_dir(pkg_root, prettify_local(transformers, exclude_files))
+  changed <- withr::with_dir(pkg_root,
+    prettify_local(transformers, filetype, exclude_files)
+  )
   invisible(changed)
 }
 
-prettify_local <- function(transformers, exclude_files) {
-  r_files <- dir(
-    path = "R", pattern = "[.][rR]$", recursive = TRUE, full.names = TRUE
-  )
+prettify_local <- function(transformers, filetype, exclude_files) {
 
-  r_files <- grep("/RcppExports[.]R$", r_files, invert = TRUE, value = TRUE)
-  test_files <- dir(
-    path = "tests/testthat", pattern = "[.][rR]$",
-    recursive = TRUE, full.names = TRUE
-  )
+  filetype <- set_and_assert_arg_filetype(filetype)
+  r_files <- test_files <- data_raw_files <- vignette_files <- readme <- NULL
+
+  if ("\\.r" %in% filetype) {
+    r_files <- dir(
```

Comment:
The `paths` argument to `dir()` seems to be vectorized, not sure about older versions of R.

Reply:
Thanks - I had overlooked this. Based on [this source code from 2000](https://github.com/wch/r-source/blob/92142731348d937ec347a56a8b68d34e9dc1d172/src/library/base/man/list.files.Rd), I think it's safe to assume that the paths argument has been vectorized since pre `R 2.0` days, so I'll go ahead and combine these.